tests/LSP6KeyManager/tests/PermissionsChangeAddURD.test.ts

```diff
@@ -0,0 +1,1353 @@
+            .withArgs(
+              canOnlyAddUniversalReceiverDelegate.address,
+              "CHANGEUNIVERSALRECEIVERDELEGATE"
+            );
+        });
+      });
+
+      describe("when caller is an address with CHANGEUniversalReceiverDelegate permission", () => {
+        it("shouldn't be allowed to ADD another UniversalReceiverDelegate key", async () => {
+          const payloadParam = {
+            dataKey: universalReceiverDelegateKey1,
+            dataValue: universalReceiverDelegateA.address,
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await expect(
+            context.keyManager
+              .connect(canOnlyChangeUniversalReceiverDelegate)
+              .execute(payload)
+          )
+            .to.be.revertedWithCustomError(context.keyManager, "NotAuthorised")
+            .withArgs(
+              canOnlyChangeUniversalReceiverDelegate.address,
+              "ADDUNIVERSALRECEIVERDELEGATE"
+            );
+        });
+
+        it("should be allowed to CHANGE the UniversalReceiverDelegate key set", async () => {
+          const payloadParam = {
+            dataKey: ERC725YKeys.LSP1.LSP1UniversalReceiverDelegate,
+            dataValue: universalReceiverDelegateD.address,
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager
+            .connect(canOnlyChangeUniversalReceiverDelegate)
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+
+          expect(ethers.utils.getAddress(result)).to.equal(
+            payloadParam.dataValue
+          );
+        });
+
+        it("should be allowed to CHANGE (remove) the UniversalReceiverDelegate key set", async () => {
+          const payloadParam = {
+            dataKey: ERC725YKeys.LSP1.LSP1UniversalReceiverDelegate,
+            dataValue: "0x",
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager
+            .connect(canOnlyChangeUniversalReceiverDelegate)
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+
+          expect(result).to.equal(payloadParam.dataValue);
+        });
+      });
+
+      describe("when caller is an address with SUPER_SETDATA permission", () => {
+        before(async () => {
+          // Adding a data key by the owner to test if address with SUPER_SETDATA
```

Comment:
Is it not this instead?

```suggestion
          // Adding a LSP1 data key by the owner to test if address with SUPER_SETDATA
```